bsp/esp32_p4_function_ev_board/esp32_p4_function_ev_board.c

```diff
@@ -434,10 +577,25 @@ static lv_display_t *bsp_display_lcd_init(const bsp_display_cfg_t *cfg)
             .swap_bytes = (BSP_LCD_BIGENDIAN ? true : false),
 #endif
             .sw_rotate = cfg->flags.sw_rotate, /* Only SW rotation is supported for 90° and 270° */
+#if CONFIG_BSP_DISPLAY_LVGL_FULL_REFRESH
```

Comment:
I think, we should set buff_dma = false, when full refresh is enabled, because we haven't got enough memory.

Reply:
Yes, that was my oversight.

Reply:
I have reviewed this part, and I think it doesn’t need to be modified because, after enabling the tear prevention feature, no additional buffer will be requested. Instead, it will use the buffer allocated by the LCD, so the allocation method is not effective.

Reply:
I tried it again and I found, where is the problem. When SW rotation is enabled and `full_refresh` is enabled too, the buffer for SW rotation must be full too and there is not enough memory in DMA (capabilities of rotation buffer are same as selected for draw buffer).

Reply:
![image](https://github.com/user-attachments/assets/ed5a2e16-664c-4cd9-a6d4-313e2b9c89aa)
This was an oversight. In fact, in both modes (direct-mode and full-refresh), it is not possible to use the lv_display_rotate_area(display, &area) interface for rotation. Users will have to handle full-frame rotation themselves.

Reply:
ok, then we should disable `.sw_rotate = false,` in LVGL9 in direct and full_refresh mode.

Reply:
Okay, I have made changes according to the situation. Please check if this is suitable. In fact, this limitation has existed since v8.

Reply:
Great, the issue has been resolved! Thank you very much.